nfc/libs/NfcCoreLib/lib/Nci/phNciNfc_ListenNfcDep.c

```diff
@@ -45,21 +46,46 @@ phNciNfc_NfcDepLstnRdrAInit(
                 pRemDevInf->tRemoteDevInfo.NfcIP_Info.Nfcip_Active = 0; /* Passive communciation */
             break;
         }
+
         /* Obtain the length of Activation parameters from pBuff */
         ActvnParamsLen = pBuff[7+RfTechSpecParamsLen+PH_NCINFCTYPES_DATA_XCHG_PARAMS_LEN];
-
-        /* Holds ATR_RES if remote device is a P2P target and ATR_REQ if remote device is a
-           P2P initiator */
         if(0 != ActvnParamsLen)
         {
-            pRemDevInf->tRemoteDevInfo.NfcIP_Info.bATRInfo_Length = (ActvnParamsLen-1);
-            pRfNtfBuff = &(pBuff[7+RfTechSpecParamsLen+PH_NCINFCTYPES_DATA_XCHG_PARAMS_LEN+2]);
-            phOsalNfc_SetMemory(pRemDevInf->tRemoteDevInfo.NfcIP_Info.aAtrInfo,0,
+            AtrLen = pBuff[7+RfTechSpecParamsLen+PH_NCINFCTYPES_DATA_XCHG_PARAMS_LEN+1];
+
+            /* Holds ATR_RES if remote device is a P2P target and ATR_REQ if remote device is a
+               P2P initiator */
+            /* The relevant ATR Length is present in the first activation parameter.
+               In the case of NCI 2.0, we ignore the Data Exchange Length Reduction parameter for now */
+            if (0 != AtrLen)
+            {
+                pRemDevInf->tRemoteDevInfo.NfcIP_Info.bATRInfo_Length = AtrLen;
+                pRfNtfBuff = &(pBuff[7+RfTechSpecParamsLen+PH_NCINFCTYPES_DATA_XCHG_PARAMS_LEN+2]);
+                phOsalNfc_SetMemory(pRemDevInf->tRemoteDevInfo.NfcIP_Info.aAtrInfo, 0,
+                                    PH_NCINFCTYPES_ATR_MAX_LEN);
+                if (pRemDevInf->tRemoteDevInfo.NfcIP_Info.bATRInfo_Length <= PH_NCINFCTYPES_ATR_MAX_LEN)
+                {
+                    phOsalNfc_MemCopy(pRemDevInf->tRemoteDevInfo.NfcIP_Info.aAtrInfo,
+                                pRfNtfBuff, pRemDevInf->tRemoteDevInfo.NfcIP_Info.bATRInfo_Length);
+                }
+                else
+                {
+                    pRemDevInf->tRemoteDevInfo.NfcIP_Info.bATRInfo_Length = 0;
+                    PH_LOG_NCI_CRIT_STR("Invalid ATR_INFO length");
+                    status = NFCSTATUS_FAILED;
+                }
+            }
+        }
+
+        if(0 != RfTechSpecParamsLen)
+        {
+            pRemDevInf->tRemoteDevInfo.NfcIP_Info.bATRInfo_Length = pRfNtfBuff[0];
+            phOsalNfc_SetMemory(pRemDevInf->tRemoteDevInfo.NfcIP_Info.aAtrInfo, 0,
                                 PH_NCINFCTYPES_ATR_MAX_LEN);
             if(pRemDevInf->tRemoteDevInfo.NfcIP_Info.bATRInfo_Length <= PH_NCINFCTYPES_ATR_MAX_LEN)
             {
                 phOsalNfc_MemCopy(pRemDevInf->tRemoteDevInfo.NfcIP_Info.aAtrInfo,
-                            pRfNtfBuff,pRemDevInf->tRemoteDevInfo.NfcIP_Info.bATRInfo_Length);
+                            pRfNtfBuff + 1, pRemDevInf->tRemoteDevInfo.NfcIP_Info.bATRInfo_Length);
```

Comment:
This +1 is coming from a difference between P2P in active or in passive mode in the NCI2x specification.
In Active mode specific parameters for NFC-ACM Poll/Listen Mode have the following format:
- 1 byte for ATR Length (n) (Request or Response)
- n bytes for ATR (Request or Response)

Will add a comment to clarify this point.